cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,83 @@
 /// <reference types="cypress" />
 /// <reference types="../support" />
 
-describe('Settings page', () => {
-  before(() => {
+import SettingsPageObject from '../support/pages/setting.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import ProfilePageObject from '../support/pages/profile.pageObject';
+import SignInPageObject from '../support/pages/signIn.pageObject';
 
-  });
+const settingsPage = new SettingsPageObject();
+const homePage = new HomePageObject();
+const profilePage = new ProfilePageObject();
+const signInPage = new SignInPageObject();
 
-  beforeEach(() => {
+describe('Settings page', () => {
+  let user;
 
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
   it('should provide an ability to update username', () => {
+    signInPage.login(user.email, user.username, user.password);
+    settingsPage.visit();
+    settingsPage.usernameField.clear();
+    settingsPage.typeNewUsername(user.newUsername);
+    settingsPage.clickUpdateSettingBtn();
 
+    homePage.assertHeaderContainUsername(user.newUsername);
   });
 
   it('should provide an ability to update bio', () => {
+    signInPage.login(user.email, user.username, user.password);
+    settingsPage.visit();
+    settingsPage.typeNewBio(user.bio);
+    settingsPage.clickUpdateSettingBtn();
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(2500);
 
+    profilePage.assertBio(user.bio);
   });
 
   it('should provide an ability to update an email', () => {
+    signInPage.login(user.email, user.username, user.password);
+    settingsPage.visit();
+    settingsPage.emailField.clear();
```

Comment:
```suggestion
    settingsPage.emailField.clear();
```
move this method into typeNewEmail()